tests/test_lambdas/test_client.py

```diff
@@ -220,3 +247,32 @@ def test__invoke__returns_accepted_result_when_invoked_asynchronously(
 
     assert result == {"result": LambdaResult.ACCEPTED}
     assert caplog.messages == []
+
+
+def test__request__raises_error_when_response_could_not_be_read_correctly(
+    lambda_client: MagicMock, caplog: LogCaptureFixture
+) -> None:
+    response = {"StatusCode": 200, "Payload": "invalid-payload"}
+    lambda_client.invoke.return_value = response
+
+    with pytest.raises(AttributeError):  # type of error does not matter here
+        LambdaClient.request(
+            "test-func",
+            "GET",
+            "/home",
+        )
+
+    logged_record = caplog.records[0]
+    assert logged_record.message == "Invalid response received from test-func Lambda (op: /home)"
+
+
+def test__request__returns_response(
+    lambda_client: MagicMock,
+) -> None:
+    payload = rest_payload_factory(body={})
+
+    lambda_client.invoke.return_value = payload
+
+    result = LambdaClient.request("test-function", "/home", "GET")
+
+    assert result.to_dict() == Response({}, headers={}).to_dict()
```

Comment:
You should rather compare it to the real dictionary with all the keys/values instead of building a Response on your own.